components/tor/tor_profile_service_impl.cc

```diff
@@ -148,6 +166,29 @@ void TorProfileServiceImpl::OnExecutableReady(const base::FilePath& path) {
   }
 }
 
+void TorProfileServiceImpl::OnPluggableTransportReady(bool success) {
+  if (!success || !tor_launcher_factory_)
+    return;
+  OnBridgesConfigChanged();
+}
+
+void TorProfileServiceImpl::OnBridgesConfigChanged() {
+  auto config = tor::BridgesConfig::FromValue(
+                    local_state_->GetDictionary(tor::prefs::kBridgesConfig))
+                    .value_or(tor::BridgesConfig());
+  if (!tor_pluggable_transport_updater_ ||
+      !tor_pluggable_transport_updater_->IsReady()) {
+    if (config.use_bridges != tor::BridgesConfig::Usage::kNotUsed)
+      tor_pluggable_transport_updater_->Register();
+    return;
+  }
```

Comment:
Also when config changed to `tor::BridgesConfig::Usage::kNotUsed` we need to call `tor_pluggable_transport_updater_->Unregister()`

Reply:
Tor doesn't actually unregister (delete files) in the unregister method. It deletes files when the user disables tor instead. I follow the same strategy and remove the bridges at the same time.

Reply:
But you missed a case when tor is enabled and bridge config changed to `kNotUsed`. Currently those bridge related executables only get deleted when tor turned disabled in `BraveExtensionManagement::OnTorDisabledChanged`

Reply:
Done, now executables will be deleted when user turn off bridges.